rainyview/src/main/java/me/samlss/view/RainyView.java

```diff
@@ -439,112 +430,115 @@ public void release() {
     }
 
     /**
-     * To optimize performance, use recycler {@link #mRecycler}.
+     * The drop's handled task.
+     * Call handler to schedule the task.
      * */
-    private RainDrop obtainRainDrop() {
-        if (mRecycler.isEmpty()) {
-            return new RainDrop();
-        }
+    private Runnable mTask = new Runnable() {
 
-        return mRecycler.pop();
-    }
+        /**
+         * To optimize performance, use recycler .
+         * */
+        private RainDrop obtainRainDrop() {
+            if (mRecycler.isEmpty()) {
+                return new RainDrop();
+            }
 
-    /**
-     * Recycling the drop that are no longer in use.
-     * */
-    private void recycle(RainDrop rainDrop) {
-        if (rainDrop == null) {
-            return;
+            return mRecycler.removeFirst();
         }
 
-        if (mRecycler.size() >= mRainDropMaxNumber) {
-            mRecycler.pop();
-        }
+        /**
+         * Recycling the drop that are no longer in use.
+         * */
+        private void recycle(RainDrop rainDrop) {
+            if (rainDrop == null) {
+                return;
+            }
 
-        mRecycler.push(rainDrop);
-    }
+            if (mRecycler.size() >= mRainDropMaxNumber) {
+                mRecycler.removeFirst(); //pop
+            }
 
-    /**
-     * The drop's handled task.
-     * Call handler to schedule the task.
-     * */
-    private Runnable mTask = new Runnable() {
-        @Override
-        public void run() {
-            createRainDrop();
-            updateRainDropState();
-            mHandler.postTask(this, 20);
+            mRecycler.push(rainDrop); //push
         }
-    };
 
-    /**
-     * Now create a random raindrop.
-     * */
-    private void createRainDrop() {
 
-        if (mRainDrops.size() >= mRainDropMaxNumber || mRainRect.isEmpty()) {
-            return;
-        }
+        /**
+         * Now create a random raindrop.
+         * */
+        private void createRainDrop() {
```

Comment:
this method throws IllegalArgumentException in a specific condition, therefore add "throws" clause to the method signature.

Reply:
Done